apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/TableSchemaExtensions.cs

```diff
@@ -40,5 +40,38 @@ internal static class TableSchemaExtensions
             }
             return ret;
         }
+
+        internal static string BuildSelectedFields(this TableSchema schema)
+        {
+            if (!(schema?.Fields?.Count > 0))
+            {
+                return string.Empty;
+            }
+
+            var built = new List<string>();
+            var currentField = new List<string>();
+            BuildSelectedFields(schema.Fields, currentField, built);
+            return string.Join(",", built);
+        }
+
+        private static void BuildSelectedFields(IList<TableFieldSchema> fieldSchemas, IList<string> currentField, IList<string> built)
+        {
+            // Whatever we have in currentField is terminal.
+            // Build that an we are done.
+            if (!(fieldSchemas?.Count > 0))
+            {
+                built.Add(string.Join(".", currentField));
+            }
+            else
+            {
+                int removeAtIndex = currentField.Count;
```

Comment:
Is there any reason not to just remove the last element and *then* do the adding? (The idea is to replace the complex field with all the field paths, right?)
Either way, a comment explaining the logic would help.